scripts/encoding_pathway_analysis.R

```diff
@@ -0,0 +1,92 @@
+# Pan-Cancer Variational Autoencoder
+# Gregory Way 2017
+# scripts/encoding_pathway_analysis.R
+#
+# Performs Overrepresentation Pathway Analyses (ORA)
+#
+# Usage: Run in command line
+#
+#     Rscript --vanilla scripts/gestalt_pathway_analysis.R
+#
+# Output:
+# ORA results for:
+# 1) HGSC-specific feature encodings identified by latent space arithmetic
+# 2) SKCM distinguishing encodings
+
+library(WebGestaltR)
+library(dplyr)
+
+run_webgestalt <- function(genes, output_name, out_dir, background_file) {
+  # Function to run a WebGestalt Pathway Analysis
+  #
+  # Arguments:
+  #    genes: A character of genes to perform ORA test on
+  #    output_name: The name of the sub folder to save results in
+  #    out_dir: The directory to save all pathway analysis results in
+  #    background_file: A filepath pointing to a .txt file with a single column
+  #                     indicating the background to perform the ORA against
+  #
+  # Output:
+  #    Return overrepresented pathways as a dataframe and write full results
+  #    to the specified folder (`gestalt/output_name`)
+
+  webgestalt_output <-
+    WebGestaltR::WebGestaltR(enrichMethod = "ORA",
+                             organism = "hsapiens",
+                             interestGene = genes,
+                             interestGeneType = "genesymbol",
+                             minNum = 4,
+                             fdrMethod = "BH",
+                             is.output = TRUE,
+                             outputDirectory = out_dir,
+                             referenceGeneFile = background_file,
+                             referenceGeneType = "genesymbol",
+                             projectName = output_name)
+  return(webgestalt_output)
+}
+
+# 1) HGSC Analysis
+
+# Specify filenames
+hgsc_out_dir <- file.path("results", "pathway", "hgsc")
+bg_file <- file.path("data", "background_genes.txt")
+hgsc_files <- list.files("results", pattern = "hgsc_node", full.names = TRUE)
+
+# Perform pathway analysis
```

Comment:
Do all the HGSC files contain high weight genes only already? Just wondering why this looks different from the SKCM analyses below.

Reply:
Yeah, the format of the files are different. The HGSC files contain all the high weight genes for a specific direction. The SKCM files contain all high weight genes and a column that indicates direction.

It may be nice to standardize the file formats, but is currently beyond the scope of this PR. I filed issue #69 for future reference. I will need to do this anyway down the road!

Reply:
Sounds good